### PR TITLE
Clarification on geospatial indexing.

### PR DESCRIPTION
**Pull request by shreybatra**
1. Added documentation to use GEOSPHERE when working with latitudes and longitudes.
2. Added notes when inserting location in GEOSPHERE.
3. Added warning of deprecated geoNear command.

### CODE REVIEW

**Comment by shreybatra** [comment]
@ShaneHarvey @behackett can you guide me please?

**Comment by shreybatra** [comment]
@prashantmital @ShaneHarvey please review. How to patch evergreen?

File: doc/examples/geo.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ Geospatial Indexing Example
   client.drop_database('geo_example')
 
 This example shows how to create and use a :data:`~pymongo.GEO2D`
-index in PyMongo.
+index in PyMongo. To create a spherical (earth-like) geospatial index use :data:`~pymongo.GEOSPHERE`.
```
Comment (ShaneHarvey) [inline doc/examples/geo.rst]: Can you add ":data:`~pymongo.GEOSPHERE` instead."?

**Comment by shreybatra** [comment]
@ShaneHarvey done. Please review.

**Pull request merged**